### PR TITLE
Split up CRD manifests into their own folder

### PR DESCRIPTION
**Pull request by twz123**
This way, CRD and app deployments can be separated, e.g. if someone wishes to install Argo CD multiple times.

Also: Make update-manifests.sh fail-fast and more resilient against spaces in paths and user's CDPATH settings.

### CODE REVIEW

**Review by jessesuen — APPROVED** [review]

**Comment by jessesuen** [comment]
@twz123 

1. we have a unit test TestGenerateYamlManifestInDir which is dependent on the number of manifests in the base directory. Can you just update the number from 22 to 20 to get the unit test to pass?

2. There is an e2e test which also depends on the location of the CRD manifests. in fixture.go, this would have to be updated:
```
func (f *Fixture) setup() error {
	_, err := exec.Command("kubectl", "apply", "-f", "../../manifests/base/application-crd.yaml", "-f", "../../manifests/base/appproject-crd.yaml").Output()
```

**Review by jessesuen — CHANGES_REQUESTED** [review]
@twz123 

1. we have a unit test TestGenerateYamlManifestInDir which is dependent on the number of manifests in the base directory. Can you just update the number from 22 to 20 to get the unit test to pass?

2. There is an e2e test which also depends on the location of the CRD manifests. in fixture.go, this would have to be updated:
```
func (f *Fixture) setup() error {
	_, err := exec.Command("kubectl", "apply", "-f", "../../manifests/base/application-crd.yaml", "-f", "../../manifests/base/appproject-crd.yaml").Output()
```

**Comment by twz123** [comment]
I fixed the tests. I must have missed the GitHub notification for this one, sorry!

File: hack/update-manifests.sh [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
-SRCROOT="$( cd "$(dirname "$0")/.." ; pwd -P )"
+set -e
+
+SRCROOT="$( CDPATH='' cd -- "$(dirname "$0")/.." && pwd -P )"
 AUTOGENMSG="# This is an auto-generated file. DO NOT EDIT"
 
 update_image () {
   if [ ! -z "${IMAGE_NAMESPACE}" ]; then
-    sed -i '' 's| image: \(.*\)/\(argocd.*\)| image: '"${IMAGE_NAMESPACE}"'/\2|g' ${1}
+    sed -i -- 's| image: \(.*\)/\(argocd.*\)| image: '"${IMAGE_NAMESPACE}"'/\2|g' "${1}"
```
Comment (twz123) [inline hack/update-manifests.sh]: Not sure whether this is an BSD vs GNU thing, but the previous version failed for me on Linux. Hope this works on OSX, too?

Comment (jessesuen) [reply hack/update-manifests.sh]: Just tried it -- it works on OS X, so you're good.

**Review by jessesuen — APPROVED** [review]

File: hack/update-manifests.sh (hunk 1 above)
Comment (jessesuen) [reply hack/update-manifests.sh]: Oops I spoke too soon, I think we need to follow something like:
https://stackoverflow.com/questions/5694228/sed-in-place-flag-that-works-both-on-mac-bsd-and-linux

to support in-place editing that works on both linux and mac

**Pull request merged**